rllib/examples/envs/classes/correlated_actions_env.py

```diff
@@ -1,74 +1,79 @@
-import gymnasium as gym
-from gymnasium.spaces import Box, Discrete, Tuple
-import numpy as np
 from typing import Any, Dict, Optional
 
+import gymnasium as gym
+import numpy as np
 
-class AutoRegressiveActionEnv(gym.Env):
-    """Custom Environment with autoregressive continuous actions.
 
-    Simple env in which the policy has to emit a tuple of correlated actions.
+class CorrelatedActionsEnv(gym.Env):
+    """Environment that can only be solved through an autoregressive action model.
 
     In each step, the agent observes a random number (between -1 and 1) and has
-    to choose two actions a1 and a2.
+    to choose two actions, a1 (discrete, 0, 1, or 2) and a2 (cont. between -1 and 1).
+
+    The reward is constructed such that actions need to be correlated to succeed. It's
+    impossible for the network to learn each action head separately.
 
-    It gets 0 reward for matching a2 to the random obs times action a1. In all
-    other cases the negative deviance between the desired action a2 and its
-    actual counterpart serves as reward. The reward is constructed in such a
-    way that actions need to be correlated to succeed. It is not possible
-    for the network to learn each action head separately.
+    There are two reward components:
+    The first is the negative absolute value of the delta between 1.0 and the sum of
+    obs + a1. For example, if obs is -0.3 and a1 was sampled to be 1, then the value of
+    the first reward component is:
+    r1 = -abs(1.0 - [obs+a1]) = -abs(1.0 - (-0.3 + 1)) = -abs(0.3) = -0.3
+    The second reward component is computed as the negative absolute value
+    of `obs + a1 + a2`. For example, if obs is 0.5, a1 was sampled to be 0,
+    and a2 was sampled to be -0.7, then the value of the second reward component is:
+    r2 = -abs(obs + a1 + a2) = -abs(0.5 + 0 - 0.7)) = -abs(-0.2) = -0.2
+
+    Because of this specific reward function, the agent must learn to optimally sample
```

Comment:
Maybe describe in a sentence why this specific reward design supports the agent in learning correlated actions?